packages/origin-247-certificate/src/offchain-certificate/synchronize/blockchain-synchronize.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { CertificateEventRepository } from '../repositories/CertificateEvent/CertificateEvent.repository';
+import { SYNCHRONIZE_STRATEGY, SynchronizeStrategy } from './strategies/synchronize.strategy';
+import { CERTIFICATE_EVENT_REPOSITORY } from '../repositories/repository.keys';
+
+@Injectable()
+export class BlockchainSynchronizeService {
+    constructor(
+        @Inject(CERTIFICATE_EVENT_REPOSITORY)
+        private readonly certEventRepo: CertificateEventRepository,
+
+        @Inject(SYNCHRONIZE_STRATEGY)
+        private readonly synchronizeStrategy: SynchronizeStrategy
+    ) {}
+
+    public async synchronize() {
+        const events = await this.certEventRepo.getAllNotProcessed();
+
+        await this.synchronizeStrategy.synchronize(events);
```

Comment:
If we call it using the queue it will probably not allow triggering two synchronization at once:
```
    @Process({ concurrency: SYNCHRONIZATION_CONCURRENCY })
```